### DISCUSSION TITLE
Cannot control devices

### DISCUSSION THREAD

**Discussion by Obstein** [make-all/tuya-local#938] [General]
Hi
I have successfully added 2lights and lawn mower. I see actuall status bitllut cannot control it. After few seconds it back to earlier state

**Comment by make-all** [comment]
A few users have encountered this before.  At least one case was fixed by manually setting the protocol (to 3.2, I think, as that protocol is very similar to 3.3 and often misdetected).  You may need to remove the device and set it up again to change the protocol version, as devices seem to get stuck and cannot be reconfigured.

**Reply by Obstein** [reply]
Unfortunately can't add with other protocol than 3.3. 
On localtuya works with 3.3 but this integration is easier to setup so this is why I want to achieve it in tuya local.
It seems like to be only in read mode. Trying 2023.7 also main version

**Reply by make-all** [reply]
Another cause of read-only mode was an incorrect device id IIRC.  It seems some devices don't check the device id before replying with status, but they do check it before acting on commands.  With 3.1 protocol devices it can also be an incorrect local key, as the encryption is only used for commands, but with 3.3 the status is also encrypted so it won't be that.

Protocol version 3.2 is very similar to 3.3, and sometimes devices can seem to partially work as 3.3, but only fully work when set to 3.2, so that is another possibility.

**Reply by Obstein** [reply]
Yeah now it works when I use device id instead uuid. Thanks
